org2jekyll.el

```diff
@@ -152,31 +172,52 @@ POST-TITLE is the title.
 POST-DESCRIPTION is the description.
 POST-TAGS is the tags
 POST-CATEGORIES is the categories."
-  (format org2jekyll-jekyll-org-post-template blog-layout blog-author post-date (org2jekyll--yaml-escape post-title) post-description post-tags post-categories))
+  (format org2jekyll-jekyll-org-post-template blog-type blog-layout blog-author post-date (org2jekyll--yaml-escape post-title) post-description post-tags post-categories))
 
 (defun org2jekyll--draft-filename (draft-dir title)
   "Compute the draft's filename from the DRAFT-DIR and TITLE."
   (concat draft-dir (org2jekyll--make-slug title) org2jekyll-jekyll-post-ext))
 
+(defsubst org2jekyll--get-post-types ()
+  "Return a list of post types.
+
+3 sources are queried, the result is uniqfied."
+  (let ((predefined-types '("page" "post"))
+        (prj-types (--keep (when (string= (plist-get (cdr it) :base-extension)
+                                          "org")
+                             (car it))
+                           org-publish-project-alist)))
+    (-uniq (append predefined-types prj-types org2jekyll-jekyll-draft-user-types))))
+
 ;;;###autoload
 (defun org2jekyll-create-draft ()
   "Create a new Jekyll blog post with TITLE.
 The `'%s`' will be replaced respectively by the blog entry name, the author, the
  generated date, the title, the description, the tags and the categories."
   (interactive)
-  (let ((author      org2jekyll-blog-author)
-        (date        (org2jekyll-now))
-        (layout      (ido-completing-read "Layout: " '("post" "default") nil 'require-match))
-        (title       (read-string "Title: "))
-        (description (read-string "Description: "))
-        (tags        (read-string "Tags (csv): "))
-        (categories  (read-string "Categories (csv): ")))
-    (let ((draft-file (org2jekyll--draft-filename (org2jekyll-input-directory org2jekyll-jekyll-drafts-dir) title)))
-      (if (file-exists-p draft-file)
-          (find-file draft-file)
+  (let* ((author      org2jekyll-blog-author)
+         (date        (org2jekyll-now))
+         (type        (ido-completing-read "Type: "
+                                           (org2jekyll--get-post-types)
+                                           nil 'require-match))
+         ;; hard-coded layouts, the general idea is that the user need to
+         ;; customize it on their own.
+         (layout      (if (string= type "page") "default" "post"))
```

Comment:
Type here is collection as explained in other line note.
Nope. They should not be related. A main change necessary to support collection is to separate these two. Currently org2jekyll uses layout to determine whether an article is a page or post. With more custom types/collections, this is no longer feasible.

Further "layout" is more about styles(representations), "type/collection" is about category/theme and thus they should not be mixed.

On the other hand, some collection might have its own layout setup, but this is what users define.